packages/babel-plugin/src/utils/state-manager.js

```diff
@@ -159,9 +159,28 @@ export default class StateManager {
   }
 
   get runtimeInjection(): ?$ReadOnly<{ from: string, as?: string }> {
-    return typeof this.options.runtimeInjection === 'string'
-      ? { from: this.options.runtimeInjection }
-      : this.options.runtimeInjection || null;
+    const options = this.options || {};
+    const runtimeInjection = options.runtimeInjection;
+    const rootDir = this.options.unstable_moduleResolution.rootDir || '';
+    const filename = this.filename || '';
+
+    if (typeof runtimeInjection === 'string' && rootDir && filename) {
+      if (
+        runtimeInjection.startsWith('./') ||
+        runtimeInjection.startsWith('../')
+      ) {
+        const absolutePath = path.join(rootDir, runtimeInjection);
```

Comment:
I think should add a normalize function to platform differences.

Reply:
you mean path.normalize(path.join()) ?

Reply:
perhaps you can refer https://github.com/vitejs/vite/blob/5684fcd8d27110d098b3e1c19d851f44251588f1/packages/vite/src/node/utils.ts#L231-L233